### PR TITLE
Update Nexus Scripts for public GoMePubNew

### PR DESCRIPTION
**Pull request by DoobesURU**
I believe these should be set for the H'uru side of things given the scripts are different from MOULa, but they should probably get another look just in case.

### CODE REVIEW

File: Scripts/Python/nxusBookMachine.py [hunk 1]
```diff
@@ -203,9 +203,9 @@
 }
 
 
-kHiddenPersonalAges = ["Personal", "Nexus", "Neighborhood", "city", "AvatarCustomization", "Cleft", "BaronCityOffice", "BahroCave", "PelletBahroCave", "Kveer", "Myst", "LiveBahroCaves", "LiveBahroCave", "Tiam"]
+kHiddenPersonalAges = ["Personal", "Nexus", "Neighborhood", "city", "AvatarCustomization", "Cleft", "BaronCityOffice", "BahroCave", "PelletBahroCave", "Kveer", "Myst", "LiveBahroCaves", "LiveBahroCave", "GoMePubNew", "ChisoPreniv", "VeeTsah", "Serene", "Vothol", "trebivdil"]
 kHiddenCityLinks = ["islmPalaceBalcony03", "KadishGallery", "islmPalaceBalcony02", "islmDakotahRoof", ]
-kHiddenAgesIfInvited = ["BahroCave", "PelletBahroCave", "Pellet Cave", "LiveBahroCave", "LiveBahroCaves", "Myst", "Tiam"]
+kHiddenAgesIfInvited = ["BahroCave", "PelletBahroCave", "Pellet Cave", "LiveBahroCave", "LiveBahroCaves", "Myst", "GoMePubNew"]
```
Comment (Hoikas) [inline Scripts/Python/nxusBookMachine.py:208]: There is no linking panel for the Age "Tiam."

File: Scripts/Python/nxusBookMachine.py [hunk 2]
```diff
@@ -236,6 +238,7 @@
      'GuildPub-Maintainers' : ("Nexus.Messages.GuildPubFull", "Nexus.Messages.GuildPubPopulation"),
      'GuildPub-Messengers' : ("Nexus.Messages.GuildPubFull", "Nexus.Messages.GuildPubPopulation"),
      'GuildPub-Writers' : ("Nexus.Messages.GuildPubFull", "Nexus.Messages.GuildPubPopulation"),
+     'GoMePubNew' : ("Nexus.Messages.GoMePubNewFull", "Nexus.Messages.GoMePubNewPopulation"),
```
Comment (Hoikas) [inline Scripts/Python/nxusBookMachine.py:241]: Does `Nexus.Messages.GoMePubNewFull` exist in the Nexus LOC files already? I don't see a PR for adding this message.

File: Scripts/Python/nxusBookMachine.py [hunk 3]
```diff
@@ -1297,6 +1310,9 @@ def IUpdateCityLinksList(self):
             #special case: and another one for GuildPub name
             elif hasattr(ageData, 'guild'):
                 displayName = "The %s' Pub" % (ageData.guild)
+            #another special case for the new GoMe Pub
+            elif ageData.ageFilename == 'GoMePubNew':
+                displayName = "New Messengers Pub"
```
Comment (Hoikas) [inline Scripts/Python/nxusBookMachine.py:1315]: I think we need to remove this in favor of... well, something not hardcoded.

File: Scripts/Python/nxusBookMachine.py (hunk 1 above)
Comment (DoobesURU) [reply Scripts/Python/nxusBookMachine.py]: Was the Nexus not updated? I know @Hazado was working on that for this release. That version should have the Linking panels, which is why we removed Tiam from the list.

Also note that I just updated these lists to match what's on MOULa.

File: Scripts/Python/nxusBookMachine.py (hunk 2 above)
Comment (DoobesURU) [reply Scripts/Python/nxusBookMachine.py:241]: Hrm...I think I did add that somewhere. Need to recheck.

File: Scripts/Python/nxusBookMachine.py (hunk 1 above)
Comment (Hoikas) [reply Scripts/Python/nxusBookMachine.py]: H-uru/moul-assets#104 is blocked due to unknown linking panels being included in the PRPs, which is inappropriate for mainline.

File: Scripts/Python/nxusBookMachine.py (hunk 3 above)
Comment (DoobesURU) [reply Scripts/Python/nxusBookMachine.py:1315]: Sure! If you have another suggestion, I'll be happy to change it.

File: Scripts/Python/islmRegisterNexusLink.py [hunk 4]
```diff
@@ -108,6 +108,20 @@ def OnNotify(self,state,id,events):
...
+                if entry is not None:
+                    entryValue = entry.chronicleGetValue()
+                    if entryValue != "yes":
+                        entry.chronicleSetValue("yes")
+                        entry.save()
+                        PtDebugPrint("islmRegisterNexusLink.OnNotify(): Chronicle entry 'GotLinkToGoMePublic' already added, setting to 'yes'")
+                else:
+                    vault.addChronicleEntry("GotLinkToGoMePublic",1,"yes")
+                    PtDebugPrint("islmRegisterNexusLink.OnNotify(): Chronicle entry 'GotLinkToGoMePublic' not present, adding entry and setting to 'yes'")
+                    PtSendKIMessage(kKILocalChatStatusMsg,PtGetLocalizedString("KI.Messages.NexusLinkAdded"))
```
Comment (Hoikas) [inline Scripts/Python/islmRegisterNexusLink.py:124]: In a future update, I would like to change this script to make this and the K'veer case more generic. It will require simultaneous updates to K'veer and GoMePubNew, however. For now, we probably just need to go with this since it got released to MOULa without any proper code review.

File: Scripts/Python/nxusBookMachine.py (hunk 3 above)
Comment (Hoikas) [reply Scripts/Python/nxusBookMachine.py:1315]: Can we add an entry to a loc file and use `PtGetLocalizedString()`?

**Comment by Hoikas** [comment]
Please rebase this on top of the latest master to resolve the failures.

File: Scripts/Python/nxusBookMachine.py (hunk 1 above)
Comment (DoobesURU) [reply Scripts/Python/nxusBookMachine.py]: > [H-uru/moul-assets#104](https://github.com/H-uru/moul-assets/pull/104) is blocked due to unknown linking panels being included in the PRPs, which is inappropriate for mainline.

I saw the PR [H-uru/moul-assets#104](https://github.com/H-uru/moul-assets/pull/104) was merged. Which panels were included and excluded? What's the latest advice for this portion of the script?

File: Scripts/Python/nxusBookMachine.py (hunk 2 above)
Comment (DoobesURU) [reply Scripts/Python/nxusBookMachine.py:241]: PR https://github.com/H-uru/moul-assets/pull/130 should properly update the Nexus LOCs for the Pub messages (at least I hope they do after proper review!).

File: Scripts/Python/nxusBookMachine.py (hunk 3 above)
Comment (DoobesURU) [reply Scripts/Python/nxusBookMachine.py]: > Can we add an entry to a loc file and use `PtGetLocalizedString()`?

Done (I think). Also added the message to the Nexus LOCs in the associated PR on moul-assets.

File: Scripts/Python/nxusBookMachine.py [hunk 5]
```diff
@@ -1297,6 +1310,9 @@ def IUpdateCityLinksList(self):
             #special case: and another one for GuildPub name
             elif hasattr(ageData, 'guild'):
                 displayName = "The %s' Pub" % (ageData.guild)
+            #another special case for the new GoMe Pub
+            elif ageData.ageFilename == 'GoMePubNew':
+                displayName = PtGetLocalizedString("Nexus.Messages.GoMePubNewName")
```
Comment (Hoikas) [inline Scripts/Python/nxusBookMachine.py:1315]: You'll probably want to put this under `Global.AgeNames.GoMePubNew` 😉

Comment (DoobesURU) [reply Scripts/Python/nxusBookMachine.py:1315]: > You'll probably want to put this under `Global.AgeNames.GoMePubNew` 😉

Ah, didn't even know that was a thing. 😊  I'll make the change.

File: Scripts/Python/nxusBookMachine.py [hunk 6]
```diff
@@ -203,9 +203,9 @@
 }
 
 
-kHiddenPersonalAges = ["Personal", "Nexus", "Neighborhood", "city", "AvatarCustomization", "Cleft", "BaronCityOffice", "BahroCave", "PelletBahroCave", "Kveer", "Myst", "LiveBahroCaves", "LiveBahroCave", "Tiam"]
+kHiddenPersonalAges = ["Personal", "Nexus", "Neighborhood", "city", "AvatarCustomization", "Cleft", "BaronCityOffice", "BahroCave", "PelletBahroCave", "Kveer", "Myst", "LiveBahroCaves", "LiveBahroCave"]
```
Comment (Hoikas) [inline Scripts/Python/nxusBookMachine.py:206]: Let's leave this for a separate PR.

File: Scripts/Python/nxusBookMachine.py [hunk 7]
```diff
@@ -203,9 +203,9 @@
...
 kHiddenCityLinks = ["islmPalaceBalcony03", "KadishGallery", "islmPalaceBalcony02", "islmDakotahRoof", ]
-kHiddenAgesIfInvited = ["BahroCave", "PelletBahroCave", "Pellet Cave", "LiveBahroCave", "LiveBahroCaves", "Myst", "Tiam"]
+kHiddenAgesIfInvited = ["BahroCave", "PelletBahroCave", "Pellet Cave", "LiveBahroCave", "LiveBahroCaves", "Myst"]
```
Comment (Hoikas) [inline Scripts/Python/nxusBookMachine.py:208]: Let's leave this for a separate PR.

**Comment by DoobesURU** [comment]
Both hidden Age lists have been reverted. Will put to a new PR later.

**Review by Hoikas — APPROVED** [review]
